.github/workflows/largemodel_unit_test_CI.yml

```diff
@@ -162,8 +164,59 @@ jobs:
           export PRIVATE_MODEL_TESTS_AWS_SECRET_ACCESS_KEY=${{ secrets.PRIVATE_MODEL_TESTS_AWS_SECRET_ACCESS_KEY }}
           export PRIVATE_MODEL_TESTS_HF_TOKEN=${{ secrets.PRIVATE_MODEL_TESTS_HF_TOKEN }}
           
-          export PYTHONPATH="./marqo/tests:./marqo/src:./marqo"
-          pytest marqo/tests --largemodel --ignore=marqo/tests/test_documentation.py --ignore=marqo/tests/backwards_compatibility_tests 
+          cd marqo
+          export PYTHONPATH="./tests:./src:."
+          set -o pipefail
+          pytest --largemodel --ignore=tests/test_documentation.py --ignore=tests/backwards_compatibility_tests \
+            --durations=100 --cov=src --cov-branch --cov-context=test \
+            --cov-report=html:cov_html --cov-report=xml:cov.xml --cov-report term:skip-covered \
+            --md-report --md-report-flavor gfm --md-report-output pytest_result_summary.md \
+            tests | tee pytest_output.txt
+
+      - name: Check Test Coverage of New Code
+        id: check_test_coverage
+        continue-on-error: true
```

Comment:
what error do we forsee could happen in this job?

Reply:
I used to have fail-on percentage in this task, but removed it. It can still fail when running diff-cover command. And we still want to upload the cov report generated from the last step even if this step fails